### DISCUSSION TITLE
The future of Objection.js

### DISCUSSION THREAD

**Discussion by koskimas** [Vincit/objection.js#2463] [Announcements]
Hello everyone!

As you may have noticed, the development of Objection has been very slow this year. That's because I've been the only active long-term developer since I created this project and I just haven't worked on it lately.

The main reason for this is that Objection's typescript typings would need a full rewrite for them to become competitive and up to today's high standards for typescript codebases. I simply don't have time, or quite frankly the interest, to do that. Objection was designed before typescript was widely adopted and I made design choices that, even today, are very difficult to express using typescript.

This puts Objection in a difficult spot since there is no other active developer that could take over. Even if I found one, it would take a long time and a large amount of contributions before I could trust that person enough to give him/her access to Objection's npm project. Realistically, I just don't see this happening. Maybe someone from Vincit could step in 😉

I understand that Objection has relatively large amount of users and projects that depend on it. But  since I left Vincit several years ago, I've maintained Objection on my spare time for free. This is the problem with OSS development. The bus factor is high. Nobody is getting paid.

I'm happy to transfer this project to someone else's care if that can somehow be arranged, but other than that, I won't be working on Objection anymore.

**Comment by devinivy** [comment]
Always appreciate all your work on Objection, Sami— needless to say it's a fantastic take on the ORM and has been one of my favorite tools in app building.  So cheers, and thanks 🍻 

I am hoping we can have some discussion here about next steps.  In my view Vincit might be the natural home for this project given the project's history, and having a clear indication from Vincit about whether they are open to that would be really useful!

We lean on Objection in hapi pal, and while I can't throw my hat in the ring for developing Objection, I intend to support whomever may take the reigns and continue to remain involved in the ecosystem!

**Comment by lehni** [comment]
I too would like to express my appreciation of the work you have done on this well crafted library @koskimas. As an OSS developer of another library (paper.js) I understand your reasoning too well, yet I am sad to read that you have decided to stop working on Objection.

I cannot judge the challenge of updating the TypeScript typings. Is there an open issue that outlines the current short-comings? I am curious… And I do wonder what an Objection rewrite in TS on top of [kysely](https://github.com/koskimas/kysely) would feel like.

Regarding the quest for future maintainers, I really hope someone can be found. I think it would be fine if the library entered maintenance mode where bugs are fixed and dependencies are kept up to date, but no or only few new major features are added. I am happy to volunteer putting in time to help out, since we have an interest in keeping this library alive for our work at Lineto.com where it is in heavy use.

Have you reached out to the folks at Vincit about this yet, Sami?

**Comment by mmmmmrob** [comment]
As a long time user of objection on several startups and projects @koskimas, you have my thanks also.

Objection is beautifully designed and written and has been an absolute joy to work with. I hope maintainers can be found and Objection can continue with a similar philosophy.

@koskimas Good luck in all you do.

**Comment by jorgebv** [comment]
Hello @koskimas, completely understandable decision that you cannot maintain such a large library for free. I have had quite a nice time with the library and thank you for maintaining it all this time.

There are a couple commits in the master branch that have never seen a release, and #2307 in particular would be nice for those hoping to ride the library off into the sunset for a little while longer. It would be nice to have this one last commit included in a release so projects can continue to use objection with the latest Typescript versions, even if Objection isn't getting further updates.

Of course if you don't feel like making any more releases, I understand. I will patch the package myself locally or make a private fork just for my project which is using Objection.

Thanks again!

**Comment by ErisDS** [comment]
First of all - thank you for all the work you've done to create & maintain Objection so far. I've done extensive research and it still seems to be the best Node.js ORM there is in terms of features and performance. It's an incredible piece of work.

Ghost has been looking to make the switch from Bookshelf for a year, but with the quietness on the repo we decided to prioritise other things and wait and see what happened for a bit. It's great to have the situation out in the open.

I am completely understanding of why you don't want to put the work in for free anymore, but I am curious, if funding existed, would that change things for you even short term? This is such a great project, it would be awesome if it could get a happy send off to new maintainers in a good state and with knowledge transfer.

I'm also curious if you've reached out to Vincit or if there's anyone there you can ping on this thread as they do seem like they would be the natural overseers... but also maybe a group could be formed?

**Comment by hkjeffchan** [comment]
Thanks for your fantastic work. I agree with @lehni to put this into maintenance mode is good enough for most of us.

**Comment by mgan59** [comment]
Just saw the Reddit thread that lead me back here.  Thanks @koskimas for all the work on ObjectionJS, I picked up the library almost 4 years ago and has been a staple in my several projects. But I totally understand the last two years Typescript is everywhere and first-class support is necessary.  Excited to see the work on Kysely and will check it out.  Best of luck! 🙌🏻

**Comment by vladshcherbin** [comment]
@koskimas thank you for all your work, being using objection and recommending it during all years of usage. I bet you even remember me from model snake/camel-case conversion request like 5 years ago (man, time runs so fast) 😅

Huge love and good luck in all your projects, totally understand your OSS pressure ♥️

**Comment by kasvith** [comment]
Thank you @koskimas for your hard work. Objection.js was the only ORM that made sense. I was using several other ORMs and jumped into Objection because it was the only one that had a really good design.

I was recommending and using it for countless projects now during my career. Its sad to see its getting obsolete. But I hope there will be a better one with Kysely 

Good luck and thanks again

**Comment by Alexigbokwe** [comment]
Thanks @koskimas for all the hard work and the creativity around Objection Js, Objection Js is amazing, and easy to use✌️

**Comment by kibertoad** [comment]
@koskimas Could I ask for npm publish permissions? I barely have time to maintain knex.js these days, but I can review and publish some fixes for Objection here and there, and I already have write permissions on the repo itself

**Reply by perrydillard** [reply]
I love both knex and Objection and would possibly be interested in joining both projects. Objection, imho, is by far the best ORM I've worked with and I've been doing SQL and OO for over 30 years across a wide range of languages and databases. I've worked with ISAM, RDB, DB2, FoxBASE, Informix, Oracle, Access, SQLServer, MySQL, SQLite, Postgres, Cloudscape, etc. and have touched most languages out there including Informix-4GL, Fortran, Pascal (Delphi), C/C++ (w/ ESQL/C), VB, MSAccess, C#, PHP, Python, Java, Javascript/Typescript and others.

**Reply by KrisLau** [reply]
@perrydillard I believe @lehni would be able to help you out with getting added as a contributor to Objection if I'm not mistaken!

**Comment by glensc** [comment]
Why not set up CI action to publish npm automatically on git tag push?

example:
- https://github.com/glensc/node-cached_property/pull/4

I could send a PR if there's interest.

**Comment by laurisvan** [comment]
@koskimas thanks also from my side - Objection.js has helped us a long way forward. I hope the libary finds a maintainer, but luckily i has not needed that much active maintenance to stay working (it is the new features/refactoring to TS that takes the effort).

Any chances of donating it to Linux Foundation or such, in spirit of Moment.js? I'm not sure if it finds a maintainer there, but definitely worth checking out.

**Comment by koskimas** [comment]
> @koskimas Could I ask for npm publish permissions? I barely have time to maintain knex.js these days, but I can review and publish some fixes for Objection here and there, and I already have write permissions on the repo itself

@kibertoad Sure! What's your npm user?

**Comment by tonyka** [comment]
Regardless of this project's future, a release needs to be tagged so that it works with the last 2 TS releases (4.8 & 4.9).

The fix is already in: #2307

**Comment by kibertoad** [comment]
@koskimas it's kibertoad. sorry, missed last invite, could you please repeat it? I'll do the release then.

**Comment by koskimas** [comment]
@kibertoad I sent a new invite

**Comment by iamnashdev** [comment]
It would be a shame to see objection going obsolete/deprecated, it is the right compromise querybuilder/orm which is really necessary when you have to deal a lot with edge cases and more complicated stuff but still want all the advantages of an orm.
Looking forward to see someone taking ownership of this projecct.
Thanks for what you have done @koskimas even though I have started using objection recently.

**Comment by lehni** [comment]
@kibertoad please let us know if you decided against taking over. In that case, we should keep looking for somebody else to take care of the maintenance. Let's not let this beautiful library wither and die.

**Comment by KrisLau** [comment]
Just saw this and I think this might be suited for https://github.com/pickhardt/maintainers-wanted (EDIT: Added!). Some other promising ones might be:

- https://www.codeshelter.co/projects/add/ (requires a current maintainer to add)
- https://github.com/garritfra/seeking-maintainers.net (EDIT: Added!)

@koskimas is there a monthly donation amount that would maybe make it viable for you to continue sustaining the project? I bet there'd be a decent number of people/companies here who would be willing to donate to keep the project maintained (esp since I saw some big names in the who uses ObjectionJS thread haha). If that's not on the table at all I completely understand and I'll probably try to hep recruit maintainers!

I just started React Native last year and either way, I have really really appreciated using ObjectionJS and truly hope that someone steps up to maintain it as it has been an absolute pleasure to use. Wishing you all the best in future endeavors!

EDIT: Not sure if this is another option to get income on the package as well: https://tidelift.com/about/lifter. I did a search for objection and it said no subscribers yet but seems like no harm in signing up. Github Sponsors would also probably be good for whoever decides to end up maintaining this full time

**Comment by flipace** [comment]
We at ovos are maintaining our [own fork of objection](https://github.com/ovos/objection.js) since quite some time. It's tricky to put time into maintenance of something that does not pay the bills, but since we're anyways working on maintaining objection for ourselves we could probably help at least with pushing new releases and some bugfixes.

It's a pity, @koskimas was always open to suggestions and was maintaining objection in such a chilled manner, really a gem.

**Comment by capaj** [comment]
If you are like me and you're looking for something typesafe, but as close to objection.js as possible, you should look at  https://orchid-orm.netlify.app/guide/

It's relatively new, not battleproven yet, but the creator is very responsive so I am confident it will get there in time.
I just want to raise awarenes about this ORM because it seems like the perfect successor to Objection. If you think it's not related to discussion in this issue let me know @koskimas and I will remove it.

**Comment by koskimas** [comment]
You should also take a look at [drizzle ORM](https://github.com/drizzle-team/drizzle-orm). It seems to be going in the same direction as objection but with amazing typescript support. And if I've understood correctly, the team behind it is VC backed.

**Comment by koskimas** [comment]
> is there a monthly donation amount that would maybe make it viable for you to continue sustaining the project

To answer this and other related questions: I've personally never wanted money for my OSS work. In my case the problem is time. I just don't have the time anymore. No amount of money would change that unless I quit my dayjob, and that seems way too risky. Any income from OSS could disappear at any time. I mentioned the lack of money in OSS because I see it as the larger problem and part of the reason why objection doesn't have more maintainers.

**Comment by rocketjsquirrel** [comment]
@koskimas , really appreciate the work you've put into this library. I have enjoyed using it on my project. I totally understand your time constraints and having to walk away. 

That said, any chance you can update the npm package as it's not sync'd with master? https://github.com/Vincit/objection.js/issues/2367

**Comment by KrisLau** [comment]
@rocketjsquirrel You might want to try using patch package until a new maintainer is found.

**Comment by rocketjsquirrel** [comment]
@KrisLau , good suggestion. I ultimately got it work referencing the package like so: 
`"objection": "https://github.com/Vincit/objection.js/archive/master.tar.gz"`

Not the most robust solution, but it'll work for now

**Comment by evilbuck** [comment]
> To answer this and other related questions: I've personally never wanted money for my OSS work. In my case the problem is time. I just don't have the time anymore. No amount of money would change that unless I quit my dayjob, and that seems way too risky. Any income from OSS could disappear at any time. I mentioned the lack of money in OSS because I see it as the larger problem and part of the reason why objection doesn't have more maintainers.

@koskimas 

I'll be blunt.
- How much money would be enough to quit your day job?
- To mitigate risk, how many months/years of paychecks stored in an escrow or some type of holding designated for payment would make this an acceptable amount of risk for you?
- Would you be willing to train someone else? We need to increase the bus factor to more than one to start.
- If this was possible, is this something you would even want to take on?

You should be compensated for your time. It's the only way this works. There should be a plan to remove you as the blocker. This should be an ongoing sponsorship or ownership of some kind to be sustainable.

**Comment by kibertoad** [comment]
Hey everyone.
I'll start addressing outstanding PRs and releasing new versions shortly. It won't be super fast, as I have other projects to tend to, but I'll do my best :)

**Comment by evilbuck** [comment]
> Hey everyone. I'll start addressing outstanding PRs and releasing new versions shortly. It won't be super fast, as I have other projects to tend to, but I'll do my best :)

I must have missed an announcement. Are you taking over?

**Comment by kibertoad** [comment]
Not taking over per se; I'm a maintainer of knex (which objection.js is using under-the-hood), and @koskimas already invited me to this repo a while ago, I just wasn't super active. Now that I have also received publish permissions, at least I can help unblock some of the more burning PRs.

I can't promise any active development from my side, though, just the general stewardship.

**Comment by kibertoad** [comment]
3.0.2 is out, please let me know if we broke something.
Next, I'll merge PR for newer Knex support and release a new semver major.

**Comment by lehni** [comment]
Thank you so much @kibertoad and @koskimas for making this happen! @kibertoad let me know if I can help out in any way.

**Comment by kibertoad** [comment]
@lehni If you could review some PRs, that would be amazing

**Comment by KrisLau** [comment]
Noticed that axios uses https://opencollective.com for fundraising so that might be a good option for Objection.js as well! I'll keep an eye out for more ways to get more maintainers for the package as well

**Comment by koskimas** [comment]
> I'll be blunt.
> 
> * How much money would be enough to quit your day job?

![alot](https://user-images.githubusercontent.com/846508/229159377-7b897461-c240-42d2-a36f-6cf542002937.jpeg)

**Comment by evilbuck** [comment]
I'll take that as you're not interested?

**Comment by KrisLau** [comment]
@evilbuck I think his reply was already in https://github.com/Vincit/objection.js/issues/2335#issuecomment-1475299722
> To answer this and other related questions: ...
> I just don't have the time anymore. No amount of money would change that unless I quit my dayjob, and that seems way too risky.

The only way to make objection.js sustainable (time wise) is to probably get a larger team of maintainers around the project. That way each person would have to dedicate less time for it individually. A consistent flow of funding would probably help with attracting that maintainer base but it's no guarantee. What were you intending on suggesting tho?

**Comment by evilbuck** [comment]
> @evilbuck I think his reply was already in [#2335 (comment)](https://github.com/Vincit/objection.js/issues/2335#issuecomment-1475299722)
> 
> > To answer this and other related questions: ...
> > I just don't have the time anymore. No amount of money would change that unless I quit my dayjob, and that seems way too risky.
> 
> The only way to make objection.js sustainable (time wise) is to probably get a larger team of maintainers around the project. That way each person would have to dedicate less time for it individually. A consistent flow of funding would probably help with attracting that maintainer base but it's no guarantee. What were you intending on suggesting tho?

That comment didn't answer the question. I based my questions on that statement. Given the stipulation that he would need to quit his day job, I wanted to know what it would take to quit his day job.
I agree that we need to spread the load (bus factor) to be maintainable. Which is one of the points in my original comment. 
> I'll be blunt.
>
>How much money would be enough to quit your day job?
>To mitigate risk, how many months/years of paychecks stored in an escrow or some type of holding designated for payment would make this an acceptable amount of risk for you?
>Would you be willing to train someone else? We need to increase the bus factor to more than one to start.
>If this was possible, is this something you would even want to take on?
>You should be compensated for your time. It's the only way this works. There should be a plan to remove you as the blocker. This should be an ongoing sponsorship or ownership of some kind to be sustainable.

I'd suggest forming an org of some kind and depositing the capital required to cover salaries + other overhead. Whether it's sponsorship or fundraising. There may already be OSS vehicles for this sort of thing. I honestly haven't looked into it much. At least cover @koskimas compensation as well as other maintainers. Have enough in a war chest to cover salaries for a reasonable length of time deemed acceptable by @koskimas to remove the "riskiness" from it. 
It would be an ongoing effort. It might be a non-profit at first, but ideally, it would be self-sufficient and be able to operate with a profitable business model for longevity.

Of course, this is all moot if @koskimas isn't interested in actually pursuing this, regardless of removing risk.

**Comment by koskimas** [comment]
> I'll take that as you're not interested?

I just don't see it happening. If you look at the few people that make a good living maintaining OSS project like the absolute OSS superstar Sindre Sorhus, you'll see that they maintain hundreds of projects that are waaaaay more popular than objection. It makes zero sense to think I would be able to earn enough maintaining just objection.

And even if that was somehow possible (it's not) I don't know if I'd still be interested. I love my current job.

**Comment by evilbuck** [comment]
> And even if that was somehow possible (it's not) I don't know if I'd still be interested. I love my current job.

thanks for the response. I was just trying to help solve the problem at hand. If you're not interested, it's a hard sell for something you ultimately don't want to do. Thanks for all your hard work and effort.

**Comment by koskimas** [comment]
> > And even if that was somehow possible (it's not) I don't know if I'd still be interested. I love my current job.
> 
> thanks for the response. I was just trying to help solve the problem at hand. If you're not interested, it's a hard sell for something you ultimately don't want to do. Thanks for all your hard work and effort.

Thank YOU for trying, I really appreciate the idea! But the facts just don't support it. There just isn't a way for small projects like objection.

**Comment by KrisLau** [comment]
Not sure if this is a good idea but it might help to include that Objection is actively searching for maintainers / lead maintainer with a link to this thread in release notes for new versions moving forward? I feel like a lot of people don't check the issues tab unless they run into something on their project

**Comment by kibertoad** [comment]
I'm not sure that's a good way to recruit good maintainer. Ideally it should be someone from the community, who has been contributing for a while already.

**Comment by koskimas** [comment]
Here's what you can do: convince your employer/people in charge to pay like 50€ a month to 10 libraries your company uses the most. That's NOTHING compared to the value you get. It might just keep the projects alive. It might make OSS development somewhat more sustainable.

**Reply by calebergh** [reply]
I'm an independent developer supporting three projects who rely heavily on Objection.js, and I would like to contribute financially.  I don't see a GitHub Sponsors or other donation mechanism set up.  What is the preferred way to support the project?

**Comment by swami-sanapathi** [comment]
Hope @koskimas finds sometime soon, until then @kibertoad is our star.

**Comment by alvinlys** [comment]
> 
> @koskimas
> 
> I'll be blunt.
> 
> * How much money would be enough to quit your day job?
> * To mitigate risk, how many months/years of paychecks stored in an escrow or some type of holding designated for payment would make this an acceptable amount of risk for you?
> * Would you be willing to train someone else? We need to increase the bus factor to more than one to start.
> * If this was possible, is this something you would even want to take on?
> 
> You should be compensated for your time. It's the only way this works. There should be a plan to remove you as the blocker. This should be an ongoing sponsorship or ownership of some kind to be sustainable.

I am unsure but why OSS communities not try to operate like subsriber-creator platform like any of media streaming (e.g. YouTube, etc.)?

**Comment by KrisLau** [comment]
@alvinlys OSS communities usually do have it (through Github Sponsors or the other options I mentioned in the thread above) but the issue they have is most people who use it don't donate or donations are insufficient to make it sustainable. Money itself is also only one problem like koskimas explained. The main problem for him (and many other open source maintainers) is time & related to that, a community of active maintainers to help reduce the burden. 

Youtube is probably a bad example to use though as it is a free entertainment platform where Google pays creators (comparitively) well even if they don't get donations. Unless GitHub/Microsoft decides to do a similar creator program, the main problem for OSS is a little different since it is entirely donation based. Plus once you post a video, it's done but if you publish a package, the issues pour in.

**Comment by lehni** [comment]
> We at ovos are maintaining our [own fork of objection](https://github.com/ovos/objection.js) since quite some time. It's tricky to put time into maintenance of something that does not pay the bills, but since we're anyways working on maintaining objection for ourselves we could probably help at least with pushing new releases and some bugfixes.
> 
> It's a pity, @koskimas was always open to suggestions and was maintaining objection in such a chilled manner, really a gem.

@flipace would you like to work together on bringing some of these features over to objection? I've looked through your repo and they look good!

I think the interesting ones are:

https://github.com/ovos/objection.js/commits/upgrade-typescript-knex-node-types
https://github.com/ovos/objection.js/commits/fix-typings-for-graphs-with-nullable-fields
https://github.com/ovos/objection.js/commits/upsertgraph-unrelate-delete-props-on-individual-models
https://github.com/ovos/objection.js/commits/v3-mod-model-type-fix
https://github.com/ovos/objection.js/commits/v3-mod-m2m-unrelate-mysql-avoid-subqueries
https://github.com/ovos/objection.js/commits/v3-mod-join-wrap-fix
https://github.com/ovos/objection.js/commits/v3-mod-allow-type-override

Did I miss anything? If you're interested, feel free to start creating merge requests for these so I can start reviewing them.

**Comment by flipace** [comment]
> > We at ovos are maintaining our [own fork of objection](https://github.com/ovos/objection.js) since quite some time. It's tricky to put time into maintenance of something that does not pay the bills, but since we're anyways working on maintaining objection for ourselves we could probably help at least with pushing new releases and some bugfixes.
> > It's a pity, @koskimas was always open to suggestions and was maintaining objection in such a chilled manner, really a gem.
> 
> @flipace would you like to work together on bringing some of these features over to objection? I've looked through your repo and they look good!
> ...
> Did I miss anything? If you're interested, feel free to start creating merge requests for these so I can start reviewing them.

@lehni I'll talk with @falkenhawk next week, and we should probably be able to create some PRs for those. A few of our changes are rather specific to our use case (such as https://github.com/ovos/objection.js/pull/14), but maybe they could still be interesting for some 👍

**Comment by koskimas** [comment]
I gave @lehni admin access to the repo as well. I've worked with him on another project and he's been contributing to objection for some years already. He's a brilliant developer and I trust him.

@lehni there's no pressure to "take over" or anything, but feel free to do so if you like/have time.

**Comment by lehni** [comment]
Thank you @koskimas for the trust, the kind words and the introduction!

Since we are heavily relying on objection.js at lineto.com, I am happy to keep helping out with maintaining the library for the foreseeable future. I don't plan on implementing any major new features, but will work on keeping things up to date and fix bugs as we encounter them. I will also keep an eye on issues and PRs and try my best to review and merge them where I see fit. I've done a first round of that over the past days, luckily there aren't that many pending issues and PRs, which I see as a sign of the quality of the library itself.

Thank you for all your hard work on this great library, and all the best with your new adventures!

**Comment by KrisLau** [comment]
Found another thing that might help with reducing workload: https://bountysource.com/

People could post their feature requests on there with a bounty. Someone picks up the issue and PRs and that way whoever picks up would only have to review it?  Still would be a lot of work with reviewing PRs but might reduce the workload on the coding side of it.

**Comment by falkenhawk** [comment]
> > We at ovos are maintaining our [own fork of objection](https://github.com/ovos/objection.js) since quite some time. It's tricky to put time into maintenance of something that does not pay the bills, but since we're anyways working on maintaining objection for ourselves we could probably help at least with pushing new releases and some bugfixes.
> > It's a pity, @koskimas was always open to suggestions and was maintaining objection in such a chilled manner, really a gem.
> 
> @flipace would you like to work together on bringing some of these features over to objection? I've looked through your repo and they look good!
> 
> (..) If you're interested, feel free to start creating merge requests for these so I can start reviewing them.

@lehni I've set up PRs for all our remaining customisations and fixes.
https://github.com/Vincit/objection.js/pull/2404 https://github.com/Vincit/objection.js/pull/2405 https://github.com/Vincit/objection.js/pull/2406 https://github.com/Vincit/objection.js/pull/2407 https://github.com/Vincit/objection.js/pull/2408 https://github.com/Vincit/objection.js/pull/2409 https://github.com/Vincit/objection.js/pull/2410

btw, would it make sense perhaps to enable [Discussions](https://github.com/Vincit/objection.js/discussions) tab for this repo?

**Reply by lehni** [reply]
@falkenhawk we have discussions now, and this issue is now one of them :)

**Comment by capaj** [comment]
I like discussions because issues often have complex templates so starting a new issue can be much more time consuming than asking some simple question in a discussion thread.

**Comment by lehni** [comment]
Great, thank you @falkenhawk! The PRs look great. I will review them soon and get back to you.

**Comment by nexdrew** [comment]
I know I'm late to the party here, but thank you @koskimas (and @kibertoad and @lehni and all other contributors) for Objection.js and all the hard work over the years! Objection.js is a great project and is really well designed. Like so many others have already stated, it's been a joy to work with.

Once I'm finally forced to use TypeScript (ugh), I will start new projects with Kysely, which appears to be knex for TypeScript.

Thanks again, and good luck to everyone!

P.S. RIP JavaScript 😢

**Reply by throrin19** [reply]
We start to rebuild all our project using typescript and objection is fully compatible with, without a problem :)

**Reply by capaj** [reply]
> We start to rebuild all our project using typescript and objection is fully compatible with, without a problem :) 

It is compatible, just doesn't give you any types for your queries so you have to type everything manually

**Reply by throrin19** [reply]
No problem for us, the query takes the type of model used, either as a basic array or as a simple object if we use fetchBy/first. 
On the other hand, as with everything, it doesn't know which columns are present in the select to get the perfect type, but a `castTo` can remedy this (and I think it's the same for all other TS ORMs).

**Comment by CosmicTiger** [comment]
Hello, I would be really interested into understand how to contribute as an active maintainer. I recently enter to this new company which I'm currently working with Knex and they use Knex raw. I wanted to use Objection for it but I feel kind of underwhelmed to propose it too because is doesn't seem currently a good catch up about a more active maintaining process (as far as I was aware from the threads, reddits posts and so forth). If I can contribute in some way or fashion helping you guys to push further this library I would love to because I could also be more confidant into propose this to my dev team as currently there's not a good typing management by itself from the current codebase.

**Reply by kibertoad** [reply]
@CosmicTiger I would honestly recommend kysely as a more modern alternative to objection.

**Reply by aryehb** [reply]
@kibertoad Isn't Kysely a query builder, more analogous to Knex, as opposed to Objection which is an ORM?

**Reply by kibertoad** [reply]
fair point. if you want orm, you may like mikroorm

**Reply by CosmicTiger** [reply]
Thank you to reply back @kibertoad appreciated it. I'll look into that then. Tysm
